### PR TITLE
[Testing] refactor: E2E param test helpers

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

- Rename `paramMap` type to `paramAnyMap` to improve readability and avoid local variable name collisions
- Add missing parameter update assertion steps
- Refactor event waiting E2E helpers

## Depends on

- #815 

## Dependents

- #820 
- #826

## Issue

Mitigates silently failing parameter update steps.

- #799

## Type of change

Select one or more from the following:

- [ ] New feature, functionality or library
- [ ] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

**Review by red-0ne — APPROVED** [review]
LGTM, preemptively approving but left a comment about a possibility to fix the `authz.MsgExec` issue.

**Review by Olshansk — APPROVED** [review]

File: e2e/tests/params_tx_test.go [hunk 1]
```diff
@@ -41,6 +41,15 @@ func (s *suite) sendAuthzExecTx(signingKeyName, txJSONFilePath string) {
 
 	// TODO_IMPROVE: wait for the tx to be committed using an events query client
 	// instead of sleeping for a specific amount of time.
+	//
+	// First attempt:
+	// eventAttrMatchFn := newEventAttributeMatchFn("action", "/cosmos.authz.v1beta1.MsgExec")
+	// s.waitForTxResultEvent(eventAttrMatchFn)
+	//
+	// This resulted in observing many more events than expected, even accounting
```
[the remark this answers is not in the document]
Comment (bryanchriswhite) [reply e2e/tests/params_tx_test.go:49]: The events that I was observing didn't contain any useful attributes to distinguish them from one another.

[the remark this answers is not in the document]
Comment (Olshansk) [reply e2e/tests/params_tx_test.go:49]: #PUC :)

[the remark this answers is not in the document]
Comment (bryanchriswhite) [reply e2e/tests/params_tx_test.go:49]: This will be removed in #826.

**Comment by Olshansk** [comment]
@bryanchriswhite What's blocking us from merging this in?

**Comment by bryanchriswhite** [comment]
> @bryanchriswhite What's blocking us from merging this in?

@Olshansk It's part of a PR tree. TL;DR, it is branched off of #815. I've updated the PR descriptions.

**Comment by bryanchriswhite** [comment]
Flaky unit test passing locally but [failing in CI](https://github.com/pokt-network/poktroll/actions/runs/10996518309/job/30573685910?pr=821):
image

**Pull request merged**